Documentation/update-modules-v3-file-api.md

```diff
@@ -0,0 +1,234 @@
+Update modules v3 protocol
+==========================
+
+Update modules are executables that are placed in `/usr/lib/mender/modules/v3`
```

Comment:
This is a great start. It would be nice to include more explanation about the purpose of the scripts and why somebody would want to create them. For example, when/why does the `Download` script run? Does it provide an alternative way to perform the download, or does it run before or after the existing Mender download? The reader first needs to understand the purpose to decide how these scripts might be relevant to their needs. Then they can proceed to the technical details.

Reply:
Agreed! I think the purpose of this doc is more architecture/technical design, though. So when it comes to actually writing modules we would need separate tutorials, potentially with references to this document.